code/modules/clothing/clothing.dm

```diff
@@ -333,16 +336,34 @@ BLIND     // can't see anything
 /obj/item/clothing/shoes/attackby(obj/item/I, mob/user, params)
 	if(istype(I, /obj/item/weapon/match) && src.loc == user)
 		var/obj/item/weapon/match/M = I
-		if(M.lit == 0)
+		if(!M.lit)
 			M.lit = 1
 			M.icon_state = "match_lit"
 			processing_objects.Add(M)
 			M.update_icon()
-			user << "<span class='notice'>You strike the [M] along your [src] to light it.</span>"
-		else if(M.lit == 1)
+			user << "<span class='warning'>You strike the [M] along [src] to light it.</span>"
+		else if(M.lit)

```

Comment:
No, leave as-is for this.

Burnt matches are lit = -1, so we have to check for lit ==1 for burning ones.